mongosql/src/translator/expressions.rs

```diff
@@ -199,17 +199,25 @@ impl MqlTranslator {
         // Separate the "normal" fields from the "set" fields. Normal fields
         // are fields that do not start with a '$' or contain a '.'.
         // Set fields are fields that start with a '$' or contain a '.'.
-        let (normal_fields, set_fields): (Vec<_>, Vec<_>) = mir_document
-            .into_iter()
-            .partition(|(k, _)| !k.starts_with('$') && !k.contains('.'));
+        // Empty fields are invalid.
+        let (normal_fields, set_fields): (Vec<_>, Vec<_>) = mir_document.into_iter().try_fold(
+            (Vec::new(), Vec::new()),
+            |(mut normal_fields, mut set_fields), (k, v)| {
+                if k.is_empty() {
+                    return Err(Error::InvalidDocumentKey(k));
+                }
```

Comment:
Sorry, I guess my previous comment was unclear. If you wanted to keep using the `partition` method to keep this more concise, that's totally fine. I just meant we don't need the `if k.is_empty()` check in both for loops (for `normal_fields` and for `set_fields`) since all empty keys would by definition be included in the `normal_fields` vector.

This current version is totally correct and fine though so no need to change unless you have a preference!